File: src/main/java/net/bloople/allblockvariants/FenceAndFenceGateCreator.java
package net.bloople.allblockvariants;

import net.minecraft.block.Block;

public class FenceAndFenceGateCreator {
    private final FenceCreator fenceCreator;
    private final FenceGateCreator fenceGateCreator;

    public FenceAndFenceGateCreator(ResourcePackBuilder builder) {
        this.fenceCreator = new FenceCreator(builder);
        this.fenceGateCreator = new FenceGateCreator(builder);
    }

    public void create(Block existingBlock) {
        fenceCreator.create(existingBlock);
        fenceGateCreator.create(existingBlock);
    }
}

File: src/main/java/net/bloople/allblockvariants/FenceAndFenceGateCreator.kt
package net.bloople.allblockvariants

class FenceAndFenceGateCreator(builder: ResourcePackBuilder) : BlockCreator(builder) {
    private val fenceCreator = FenceCreator(builder)
    private val fenceGateCreator = FenceGateCreator(builder)

    override fun create(blockInfo: BlockInfo) {
        fenceCreator.create(blockInfo)
        fenceGateCreator.create(blockInfo)
    }
}